pkg/kubelet/kubelet_pods_test.go

```diff
@@ -2083,6 +2302,10 @@ func TestPodPhaseWithRestartOnFailure(t *testing.T) {
 	}
 }
 
+// No special init-specific logic for this, see RestartAlways case
+// func TestPodPhaseWithRestartOnFailureInitContainers(t *testing.T) {
+// }
```

Comment:
I could copy-paste the test cases for RestartAlways for the RestartOnFailure cases if someone thinks there's any value... put this comment here as an explanation for why I didn't.

Reply:
I'm fine with that.